### ISSUE TITLE
Resize in responsive window fails to keep height

### ISSUE THREAD

**Issue by BackuPs** [blakewilson/vidbg#1]
Hi

Thank you for this beautifull and fast script. Finally a script that has video position top bottom correctly implemented as most scripts center the video loosing all the action in the bottom. This script allows to set a start point for the video so one can see the video from the bottom point which is great.

I found unfortunately one issue.

If you create a div box width=100% and height= 400px and set a video to the background of that box.

It does not matter if you turn off the video resize function within the script true or false as when the window hits a certain point f.e. mobiles work on (below 980px)  The video does not keep the height starts resizing and creates in ie black borders (in other browsers you get white area's) above and below the video.

In my opinion the video should keep the height of the box and center  (left right) (top bottom) the video within.

You can see a screenshot here http://i.imgur.com/Htdn4ox.png

Can you fix this?

Thanks !

**Comment by blakedotvegas** [comment]
Hi BackuPs,

Thanks for the kind words, I'm glad you enjoy it.

This usually is an issue when `video { max-width: 100%; }` is set. If this is the case, simply add the following to your stylesheet:
```
.vidbg-container video {
  max-width: none !important;
}
```

Please let me know if this solves it. Thanks!

**Comment by BackuPs** [comment]
Hi


I ma sorry i just saw your reply... Dont know why i did not get a email.

Will try and let you know this week.

Thx

**Comment by BackuPs** [comment]
Hi

You were right. There was a default max-width set at 100%.

I have one other question. If you set a bg video to a container and a fall back image the fallback image is shown shortly before the video starts. Can you please change that so the fall back image only shows on mobiles? It is confusing see this happening.

Another thing is that the video when set to full width and your screen is f.e. 1920px in IE11 you first see the video as it was designed and then it stretches out to full screen.

Is there a way to prevent this.  If you want examples pls add your email and i will send you some links with the given issues

Thanks

**Comment by blakewilson** [comment]
Hi BackuPs,

I'm glad this solved your issue!

It would be better to visualize these issues, so if you could send examples that would be perfect! You can send the email to hi@blakewilson.me . I hope to hear from you soon!

Regards,

**Comment by BackuPs** [comment]
Hi

Did you get my email with the links in it?

Regards,

**Comment by blakewilson** [comment]
Hi BackuPs,

Yes i did. I apologize I haven't gotten back to you. I have been super busy the past couple of weeks. I will try my best to resolve your issues tonight.

Again, I'm really sorry for the delay.

**Comment by BackuPs** [comment]
Hi Blake

Any idea when you are gonna resolve my issues?

Thanks

Best regards,
Paul

**Comment by blakewilson** [comment]
Hi Paul,

Although the issue with the border jumping is related to another script, I believe I have a solution that will resolve both your issues.

Try adding the following code below your init for vidbg.js:
`setTimeout(function() {
	$('.vidbg-container').css({
		'opacity': 1,
		'filter': 'alpha(opacity=1)',
		'visibility': 'visible',
	});
}, 600 );`

Also in your CSS stylesheet:
`.vidbg-container {
	opacity: 0;
	filter: alpha(opacity=0);
	visibility: hidden;
	-webkit-transition: opacity 0.5s;
	transition: opacity 0.5s;
}`


This will cause the video background to fade in after 600 milliseconds, hiding the quick flash of the fallback image, and _hopefully_ resolving the issue with the borders. 

Please let me know if this solution fixes the issue and my sincerest apologies for such a long wait for a fix.

Have a great day!

**Comment by BackuPs** [comment]
Hi Blake,

Thank you for the suggestion but that did not solve the problem.... but this did, just add width:'100%' to the video css in your script and the issue is solved in either browser


`    // Video alignment
    $video.css({
      margin: 'auto',
      position: 'absolute',
      'z-index': -1,
      top: position.y,
      left: position.x,
      '-webkit-transform': 'translate(-' + position.x + ', -' + position.y + ')',
      '-ms-transform': 'translate(-' + position.x + ', -' + position.y + ')',
      '-moz-transform': 'translate(-' + position.x + ', -' + position.y + ')',
      transform: 'translate(-' + position.x + ', -' + position.y + ')',

      // Disable visibility, while loading
      visibility: 'hidden',
      opacity: 0,
	  width:'100%'
    })
`

Best regards,
Paul

**Comment by BackuPs** [comment]
Hi Blake,

 

That did not solve the problem.... but this did, just add width:'100%' to the video css in your script and the issue is solved in either browser

 

I replied at github what the solution is.

 

Thank you for your support and sticking up with me on this.

 

Best regards,

Paul

 

 

 

Van: Blake Wilson [mailto:notifications@github.com] 
Verzonden: vrijdag 18 maart 2016 19:46
Aan: blakewilson/vidbg
CC: BackuPs
Onderwerp: Re: [vidbg] Resize in responsive window fails to keep height (#1)

 

Hi Paul,

Although the issue with the border jumping is related to another script, I believe I have a solution that will resolve both your issues.

Try adding the following code below your init for vidbg.js:
setTimeout(function() {
$('.vidbg-container').css({
'opacity': 1,
'filter': 'alpha(opacity=1)',
'visibility': 'visible',
});
}, 600 );

Also in your CSS stylesheet:
.vidbg-container {
opacity: 0;
filter: alpha(opacity=0);
visibility: hidden;
-webkit-transition: opacity 0.5s;
transition: opacity 0.5s;
}

This will cause the video background to fade in after 600 milliseconds, hiding the quick flash of the fallback image, and hopefully resolving the issue with the borders. 

Please let me know if this solution fixes the issue and my sincerest apologies for such a long wait for a fix.

Have a great day!

**Comment by ziyanziyu** [comment]
你好，视频音量设置为大于1的时候，视频画面为全白，有声音，在将音量设置为1的时候，画面正常，无声音

**Comment by jveler-ecrsoft** [comment]
Hello, I just noticed something about this. I wanted to see if I could fix this myself and so I downloaded the vidbg.js (not the minified version) and loaded it in my browser to better inspect the code. Low and behold, the responsive nature of the video worked fine. When the height of the video was less than the container, rather than scaling down, it started cutting off the left and right as I expected. Is there a difference in the vidbg.js and vidbg.min.js? 

I just minified the vidbg.js on my own to create a minified version and all works as expected.
